fem/tmop.hpp

```diff
@@ -69,6 +73,20 @@ class TMOP_QualityMetric : public HyperelasticModel
    virtual void EvalPW(const DenseMatrix &Jpt, DenseMatrix &PW) const
    { PW = 0.0;}
 
+   // First-derivative hook for AD-based computations:
+   virtual AD1Type EvalW_AD1(const std::vector<AD1Type> &T,
+                             const std::vector<AD1Type> &W) const
+   {
+      return AD1Type{0.0, 0.0};
+   }
+
+   // Second-derivative hook for AD-based computations:
+   virtual AD2Type EvalW_AD2(const std::vector<AD2Type> &T,
+                             const std::vector<AD2Type> &W) const
+   {
+      return AD2Type{{0.0, 0.0}, {0.0,0.0}};
```

Comment:
Isn't it better to abort? Now these functions can be used with any metric, and it will just work, returning zeros silently.